docs/en/stack/ml/setup.asciidoc

```diff
@@ -5,100 +5,145 @@
-roles instead and set your {kib} privileges appropriately for each space.
 
-For full access to the {ml-features} in {kib}, you must have:
+Each {ml} job can be assigned to all, one or multiple spaces. This can be 
+configured in **{stack-manage-app}** > _Alerts and Insights_ > 
+_Machine Learning Jobs_. To assign a job to a space, select the spaces icon 
+shown in the job list.
+
+[role="screenshot"]
+image::assign-job-spaces.jpg["Assign machine learning jobs to spaces"]
+
+
+[discrete]
+[[kib-security-privileges]]
+=== {kib} user
+
+Within a {kib} space, for full access to the {ml-features}, you must have:
 
 [%interactive]
-* [ ] `all` {kib} privileges for the {ml-features} in the appropriate spaces
-* [ ] `read` and `view_index_metadata` index privileges on source indices
-* [ ] `read`, `manage`, and `index` index privileges on destination indices (for
-  {dfanalytics-jobs} only)
+* [ ] `All` {kib} privileges for the {ml-features}
+* [ ] {data-sources} for your source indices
+* [ ] `read`, and `view_index_metadata` index privileges on your source indices
+* [ ] {data-sources}, `read`, `manage`, and `index` index privileges on 
+  destination indices (for {dfanalytics-jobs} only)
+
 
-For read-only access to the {ml-features} in {kib}, you must have:
+Within a {kib} space, for read-only access to the {ml-features}, you must have:
 
 [%interactive]
-* [ ] `read` {kib} privileges for the {ml-features} in the appropriate spaces
-* [ ] `read` index privileges on source indices
-* [ ] `read` index privileges on destination indices (for {dfanalytics-jobs}
-only)
+* [ ] `Read` {kib} privileges for the {ml-features}
+* [ ] {data-sources} for your source indices
+* [ ] `read` index privilege on your source indices
+* [ ] {data-sources} and `read` index privilegeson destination indices (for 
+  {dfanalytics-jobs} only)
+
+NOTE: {data-sources-cap} can be automatically created when creating a 
+{dfanalytics-job}.
 
+For access to use {ml} APIs via _Dev Tools_ in {kib}, set the {es} security 
+privileges and grant access to `machine_learning_admin` or 
+`machine_learning_user` built-in roles.
+
+
+[discrete]
 [[upload-file-security-privileges]]
-To upload and import files in the *{data-viz}*, you must have:
+=== Security privileges for {data-viz}
+
+Within a {kib} space, to upload and import files in the *{data-viz}*, you must 
+have:
 
 [%interactive]
-* [ ] `all` {kib} privileges for the {ml-app} or Discover features in
-the appropriate spaces. Alternatively, `read` {kib} privileges for the {ml-app}
-feature and `all` {kib} privileges for the {ipm-app} feature
-* [ ] `manage_pipeline` or `manage_ingest_pipelines` cluster privileges
-* [ ] `create`, `create_index`, `manage` and `read` index privileges for
-destination indices
+- [ ] `All` or `Read` Machine Learning or `All` Discover {kib} feature 
+  privileges
+- [ ] `All` {data-sources-caps} Management {kib} feature privileges (required 
+  for automatic {data-source} creation)
```

Comment:
This is actually also required to create data views via the data frame analytics wizard. In the section above we have the note `{data-sources-cap} can be automatically created when creating a 
{dfanalytics-job}.`. I'm wondering if we'd like to repeat this requirement in the note above or if we even want to extract this bit such that it work for both sections.
Any thoughts on this @szabosteve @sophiec20 ?

Reply:
Changed in 6796c74.